frontend/amundsen_application/static/js/pages/TableDetailPage/WriterDropdown/index.tsx

```diff
@@ -0,0 +1,166 @@
+// Copyright Contributors to the Amundsen project.
+// SPDX-License-Identifier: Apache-2.0
+
+import * as React from 'react';
+import { Dropdown, MenuItem } from 'react-bootstrap';
+
+import AvatarLabel from 'components/AvatarLabel';
+import { TableWriter } from 'interfaces';
+import { logClick } from 'utils/analytics';
+import {
+  APPLICATIONS_LABEL,
+  AIRFLOW,
+  DATABRICKS,
+  PRODUCING,
+  CONSUMING,
+  DAG_LABEL,
+  TASK_LABEL,
+  NONE_VALUE,
+} from './constants';
+import './styles.scss';
+
+export interface WriterDropdownProps {
+  tableApps: TableWriter[];
+}
+
+const isAirflowOrDatabricksApp = (appName) =>
+  appName.toLowerCase() === AIRFLOW || appName.toLowerCase() === DATABRICKS;
+
+const getSortedAppKinds = (apps: TableWriter[]) => {
+  const appKinds: string[] = [
+    ...new Set(apps.map((app) => (app.kind ? app.kind : PRODUCING))),
+  ];
+
+  const producingKind = appKinds.filter(
+    (kind) => kind.toLowerCase() === PRODUCING
+  );
+  const consumingKind = appKinds.filter(
+    (kind) => kind.toLowerCase() === CONSUMING
+  );
+
+  const sortedKinds: string[] = [];
+  sortedKinds.push(...producingKind, ...consumingKind);
+  sortedKinds.push(
+    ...appKinds.filter(
+      (kind) =>
+        kind.toLowerCase() !== PRODUCING && kind.toLowerCase() !== CONSUMING
+    )
+  );
+  return sortedKinds;
+};
+
+const getMenuItem = (app: TableWriter, menuItemClassName) => {
+  if (app.name.toLowerCase() === AIRFLOW) {
+    const [dagId, ...task] = app.id.split('/');
+    const taskId = task.join('/');
+    return (
+      <MenuItem
+        className={menuItemClassName}
+        href={app.application_url}
+        onClick={logClick}
+        target="_blank"
```

Comment:
Any link that has target="_blank" needs to have rel="noopener noreferrer" for security reasons.
I see this is not exactly a link, but I guess is kind of?